src/providers/ogr/qgsogrprovider.h

```diff
@@ -212,6 +298,49 @@ class QgsOgrProvider : public QgsVectorDataProvider
     //! String used to define a subset of the layer
     QString mSubsetString;
 
+    /** MSubLayerList entry that was actually used to open the layer
+     * @note
+     *
+     * After OGRGetLayerWrapper has successfully loaded a Layer
+     * an adapted version of mSubLayerList entry that was used to open the layer will be stored
+     * @see OGRGetLayerWrapper()
+     *
+     * Sample of a duplicate Layer-name created by 'subLayers':
+     * '3:Directions from 423, Taiwan, 台中市東勢區慶福里 to 423, Taiwan, Taichung City, Dongshi District, 中45鄉道:3:LineString25D:1'
```

Comment:
At the moment this sample from
* https://hub.qgis.org/issues/15168

works correctly.
But the likelihood of '|'  being used in a layename is possible.